web-api/src/business/useCases/caseDeadline/createCaseDeadlineInteractor.ts

```diff
@@ -48,15 +54,66 @@ export const createCaseDeadline = async (
     .updateCaseAndAssociations({
       applicationContext,
       authorizedUser,
-      caseToUpdate: caseEntity,
+      caseToUpdate: updatedCaseEntity,
     });
 
+  const { docketNumber, leadDocketNumber, consolidatedCases } = caseDetail;
+  if (!handlingConsolidatedCases && docketNumber === leadDocketNumber) {
+    const ADD_DEADLINE_TO_CONSOLIDATED_CASES = consolidatedCases
+      .filter(
+        ({ docketNumber: ccDocketNumber }) => ccDocketNumber !== docketNumber,
+      )
+      .map(({ docketNumber: ccDocketNumber }) => {
+        return createCaseDeadline(
+          applicationContext,
+          {
+            caseDeadline: {
+              ...caseDeadline,
+              docketNumber: ccDocketNumber,
+              consolidatedCaseDeadlineId: newCaseDeadline.caseDeadlineId,
+            } as CaseDeadline,
+            handlingConsolidatedCases: true,
+          },
+          authorizedUser,
+        );
+      });
+
+    await Promise.all(ADD_DEADLINE_TO_CONSOLIDATED_CASES);
+  }
+
   return new Case(result, { authorizedUser }).validate().toRawObject();
 };
 
+export async function getcreateCaseDeadlineLockInfo(
```

Comment:
I hate that we need to fetch the full case details just to lock. I think after this is merged we can be more granular in (or after) 10502. We really just want "all docket numbers associated with my lead docket number." This should be a simple query in Postgres.